### PR TITLE
share-utils: Add API for sharing files with SHARE permissions (GS-11286)

### PR DESCRIPTION
**Pull request by alexander-polovtcev**
Core changes - https://github.com/genestack/genestack/pull/4730

### CODE REVIEW

File: genestack_client/share_util.py [hunk 1]
```diff
@@ -10,11 +10,36 @@ class ShareUtil(Application):
...
+
+            Allows finding files via search and reading files' content
+
+        .. attribute:: EDIT
+
+            Allows finding files via search, reading files' content and modifying files' metainfo
+
+        .. attribute:: SHARE
+
+            Allows group members to share files with other groups. This is only applicable
```
Comment (genestack-solomatin) [inline genestack_client/share_util.py]: Allows group members to share files with other groups in same organization.

**Comment by alexander-polovtcev** [comment]
@genestack-solomatin please don't rush to review PRs without any assigned reviewers. I usually create PRs for my branches so that I can easily find the stuff I'm currently working on. This code is not ready for review yet.

File: genestack_client/share_util.py [hunk 2]
```diff
@@ -45,21 +74,129 @@ def share_files_for_edit(self, file_accessions, group_accession, destination_fol
...
+            self, file_accessions, group_accession, destination_folder, method, *args
+    ):
+        file_accessions = self.__to_list(file_accessions)
+        self.invoke(method, file_accessions, [group_accession], *args)
         if destination_folder is not None:
             self.invoke('linkFiles', file_accessions, destination_folder, group_accession)
 
+    def share_folder(self, folder_accession, group_accession, permissions, destination_folder=None):
+        """
+        Recursively share the given folder, its subfolders and files inside them.
```
Comment (genestack-solomatin) [inline genestack_client/share_util.py]: safe or not safe?

Comment (alexander-polovtcev) [reply genestack_client/share_util.py]: what?

**Comment by alexander-polovtcev** [comment]
> Did you check how it looks in generated documentation?
Yes

**Comment by alexander-polovtcev** [comment]
> I don't like parameter types "string or collection"

I actually think that this is quite pythonic

File: genestack_client/share_util.py (hunk 2 above)
Comment (genestack-solomatin) [reply genestack_client/share_util.py]: What will happen `if some of the given files cannot be shared by the current user (i.e. he doesn't own them or doesn't have the :attr:`ShareUtil.Permissions.SHARE` permission`

Comment (alexander-polovtcev) [reply genestack_client/share_util.py]: fixed

**Pull request merged**